osmaxx-py/osmaxx/excerptexport/services/conversion_api_client.py

```diff
@@ -0,0 +1,205 @@
+import requests
+import json
+import logging
+
+from django.core.files.base import ContentFile
+
+from osmaxx.excerptexport.models import ExtractionOrderState, OutputFile
+from osmaxx.utils import private_storage
+
+logger = logging.getLogger(__name__)
+
+
+class ConversionApiClient():
+    def __init__(self, protocol, host, port, api_paths, credentials):
+        """
+        Args:
+            protocol:       e.g. 'http'
+            host:           e.g. 'www.osmaxx.ch'
+            port:           e.g. '8000'
+            api_paths:      e.g.
+                {
+                    'login': '/api/token-auth/?format=json'
+                    'job': {
+                        'create': '/api/jobs'
+                        'status': '/api/conversion_result/{rq_job_id}'
+                    }
+                }
+            credentials:    e.g.
+                {'username':'osmaxx', 'password':'osmaxx'}
+        """
+        self.protocol = protocol
+        self.host = host
+        self.port = port
+        self.api_paths = api_paths
+        self.credentials = credentials
+        self.is_logged_in = False
+
+        self.headers = {
+            'Content-Type': 'application/json; charset=UTF-8'
+        }
+
+    def _create_url(self, path):
+            return self.protocol + '://' + self.host + ':' + (self.port if self.port else '') + path
+
+    def login(self):
+        """
+        Logs in the api client by requesting an API token
+
+        Returns:
+            True on successful login
+            False on failed login
+        """
+        request_url = self._create_url(self.api_paths['login'])
+        request_data = self.credentials
+
+        response = requests.post(request_url, data=json.dumps(request_data), headers=self.headers)
```

Comment:
instead of `if response.status_code == requests.codes.ok and json_response and 'rq_job_id' in json_response:`

I'd use something like this:

```python
def _request_successful(self, response):
        try:
            response.raise_for_status()
        except requests.HTTPError as e:
            logging.error('API job creation failed.', e.message + " response: " + response)
            return False
         return True

def _extract_key_from_json_or_None(self, response, key):
    if self._request_successful(response):
        json_response = response.json()
        return json_response.get('rq_job_id', None)
    return None
...
    rq_job_id = self._extract_key_from_json_or_None(response, rq_job_id)
    if rq_job_id:
        extraction_order.process_id = rq_job_id
...
```